### PR TITLE
Make the build not fail when `defaultUseHttp2Preface=false`

### LINKED ISSUE #1891: Confirm default of enabling http2 preface on client
Just wanted to confirm using the http2 preface is a good default for an armeria client. I ran into not being able to connect to Elasticsearch V7 when this is enabled with

`com.linecorp.armeria.client.SessionProtocolNegotiationException: expected: http, reason: connection established, but session creation timed out:`

The error message doesn't really help debug and after being stuck for a while, I had the random idea of disabling the preface, which I think most users wouldn't ever find.

If there are servers where the preface doesn't work, is it actually a good idea to default to enabling it?

For reference, had failures in this build https://travis-ci.org/openzipkin/zipkin/builds/556693286?utm_source=github_status&utm_medium=notification

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

We see some build failures when `com.linecorp.armeria.defaultUseHttp2Preface=false`
flag is set.

Modifications:

- Fix various test failures when `useHttp2Preface` is `false`.
- Fix a bug where `unfinishedResponses` becomes a negative value when
  `useHttp2Preface` is `false`.
- Miscellaneous:
  - Javadoc updates in `Flags`

Result:

- Client-side idle connection timeout works as expected when
  `useHttp2Preface` is `false`.
- Better documentation.

### CODE REVIEW

**Comment by ikhoon** [comment]
By the way, don't we need to add `breaking change` label?

**Comment by trustin** [comment]
> By the way, don't we need to add `breaking change` label?

I think it shouldn't break anything as long as the target endpoint is a proper HTTP server, so I added the `improvement` label.

**Comment by trustin** [comment]
Changed the PR subject and description, now that we are not going to change the default `defaultUseHttp2Preface`.

**Comment by trustin** [comment]
See https://github.com/line/armeria/issues/1891#issuecomment-651095440 for the background for the change.

File: core/src/main/java/com/linecorp/armeria/common/Flags.java [hunk 1]
```diff
@@ -724,7 +744,14 @@ public static int defaultHttp1MaxChunkSize() {
 
     /**
      * Returns the default value of the {@link ClientFactoryBuilder#useHttp2Preface(boolean)} option.
-     * Note that this value has effect only if a user did not specify it.
+     * If enabled, the HTTP/2 connection preface immediately for a cleartext HTTP/2 connection, reducing
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/common/Flags.java:747]: sends preface?

File: eureka/src/test/java/com/linecorp/armeria/client/eureka/ArmeriaEurekaClientTest.java [hunk 2]
```diff
@@ -63,17 +63,32 @@
...
     @Override
     protected EurekaHttpClient getEurekaClientWithBasicAuthentication(String userName, String password) {
-        final WebClient webClient = WebClient.builder(getHttpServer().getServiceURI())
+        final WebClient webClient = WebClient.builder(toH1C(getHttpServer().getServiceURI()))
                                              .auth(BasicToken.of(userName, password))
                                              .build();
         return new EurekaHttpClientWrapper(webClient);
     }
 
+    private static String toH1C(URI serviceURI) {
```
Comment (minwoox) [inline eureka/src/test/java/com/linecorp/armeria/client/eureka/ArmeriaEurekaClientTest.java:77]: Thanks!

File: core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java [hunk 3]
```diff
@@ -441,6 +441,7 @@ public void onComplete() {}
             }, ctx.channel().eventLoop());
 
             // NB: No need to set the response timeout because we have session creation timeout.
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java:443]: Let's move this comment one line down. 😄

File: core/src/main/java/com/linecorp/armeria/common/Flags.java (hunk 1 above)
Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/common/Flags.java]: Done :-)

File: core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java (hunk 3 above)
Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java]: and done!

**Pull request merged**